src/elements/earn/pools/poolsTable/v3/DepositV3Modal.tsx

```diff
@@ -43,41 +34,57 @@ import {
   getOnOff,
 } from 'services/api/googleTagManager';
 import { DepositDisabledModal } from './DepositDisabledModal';
+import { checkV3DepositWhitelist } from 'utils/v3Deposit';
+import { TradeWidgetInput } from '../../../../trade/TradeWidgetInput';
+import { useTknFiatInput } from '../../../../trade/useTknFiatInput';
+import { DepositFAQ } from './DepositFAQ';
+import { Switch, SwitchVariant } from '../../../../../components/switch/Switch';
 
 interface Props {
   pool: PoolV3;
-  renderButton: (
-    onClick: (pool_click_location?: string) => void
-  ) => React.ReactNode;
+  renderButton: (onClick: (pool_click_location?: string) => void) => ReactNode;
 }
 
-const REWARDS_EXTRA_GAS = 130_000;
-
 export const DepositV3Modal = ({ pool, renderButton }: Props) => {
-  const enableDeposit = useAppSelector((state) => state.user.enableDeposit);
+  const isBNT = bntToken === pool.poolDltId;
+  const adminEnableDeposit = useAppSelector(
+    (state) => state.user.enableDeposit
+  );
   const account = useAppSelector((state) => state.user.account);
   const [isOpen, setIsOpen] = useState(false);
   const [txBusy, setTxBusy] = useState(false);
+  const [tosAgreed, setTosAgreed] = useState(false);
   const [amount, setAmount] = useState('');
   const [inputFiat, setInputFiat] = useState('');
   const isFiat = useAppSelector((state) => state.user.usdToggle);
   const [accessFullEarnings, setAccessFullEarnings] = useState(true);
-  const [extraGasNeeded, setExtraGasNeeded] = useState('0');
-  const eth = useAppSelector((state) => getTokenById(state, ethToken));
-  const poolV3Map = useAppSelector(getPoolsV3Map);
   const { handleWalletButtonClick } = useWalletConnect();
 
+  const tokenInputField = useTknFiatInput({
+    token: pool.reserveToken,
+    setInputTkn: setAmount,
+    setInputFiat: setInputFiat,
+    inputTkn: amount,
+    inputFiat: inputFiat,
+  });
+
+  const enableDeposit =
+    adminEnableDeposit ||
+    (pool.depositingEnabled && checkV3DepositWhitelist(pool.poolDltId));
+
   const onClose = async () => {
     setIsOpen(false);
     await wait(500);
     setAmount('');
     setInputFiat('');
-    setExtraGasNeeded('0');
     setAccessFullEarnings(true);
   };
 
   const isInputError = useMemo(
-    () => !!account && new BigNumber(pool.reserveToken.balance || 0).lt(amount),
+    () =>
```

Comment:
since this now returns a string (error message) and not a boolean it should be renamed from isInputError to inputError (for example)

Reply:
done